art/metrics/metrics.py

```diff
@@ -319,21 +320,29 @@ def clever_t(
     elif norm != 2:
         raise ValueError("Norm {} not supported".format(norm))
 
-    # Loop over the batches
-    for _ in range(nb_batches):
-        # Random generation of data points
-        sample_xs = rand_pool[np.random.choice(pool_factor * batch_size, batch_size)]
+    # Compute gradients for all samples in rand_pool
+    for i in range(batch_size):
+        rand_pool_batch = rand_pool[i * pool_factor:(i+1) * pool_factor]
```

Comment:
E226 missing whitespace around arithmetic operator
```suggestion
        rand_pool_batch = rand_pool[i * pool_factor:(i + 1) * pool_factor]
```